### PR TITLE
Rename tier 2 redundancy eliminator to optimizer

### PR DESCRIPTION
**Pull request by gvanrossum**
The original name is just too much of a mouthful.

@markshannon and @Fidget-Spinner please review.

### CODE REVIEW

File: Lib/test/test_generated_cases.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def skip_if_different_mount_drives():
     import parser
     from stack import Stack
     import tier1_generator
-    import tier2_abstract_generator
+    import optimizer_generator.py
```
Comment (Eclips4) [inline Lib/test/test_generated_cases.py]:
```suggestion
    import optimizer_generator
```

File: Lib/test/test_generated_cases.py [hunk 2]
```diff
@@ -841,7 +841,7 @@ def run_cases_test(self, input: str, input2: str, expected: str):
             temp_input.flush()
 
         with handle_stderr():
-            tier2_abstract_generator.generate_tier2_abstract_from_files(
+            optimizer_generator.py.generate_tier2_abstract_from_files(
```
Comment (Eclips4) [inline Lib/test/test_generated_cases.py]:
```suggestion
            optimizer_generator.generate_tier2_abstract_from_files(
```

**Review by Eclips4 — APPROVED** [review]
LG except two typos above. Thanks!

**Comment by Fidget-Spinner** [comment]
I think Mark disagreed with optimizer at the time because there can be more than one optimization pass, and therefore multiple abstract interpreters generated. So it's not *the* tier 2 optimizer, but one of them.

**Comment by gvanrossum** [comment]
> I think Mark disagreed with optimizer at the time because there can be more than one optimization pass, and therefore multiple abstract interpreters generated. So it's not _the_ tier 2 optimizer, but one of them.

I know -- which is why I am waiting for his review. :-) I think that it will be quite a while before we have a second abstract interpreter, and we can always reconsider the naming then. Personally I find the current name way too long and also not very clear on what it actually does.

**Review by markshannon — APPROVED** [review]
Just `optimizer` does seem a bit vague, but I don't have a better name for now.

When we add another pass, we can have a rethink.

**Pull request merged**